x/gov/msg.go

```diff
@@ -124,3 +112,47 @@ func (m UpdateElectorateMsg) Validate() error {
 	}
 	return nil
 }
+
+func (CreateElectorateUpdateProposalMsg) Path() string {
+	return pathCreateElectorateUpdateProposalMsg
+}
+
+func (m CreateElectorateUpdateProposalMsg) Validate() error {
+	for i, v := range m.DiffElectors {
+		if v.Weight > maxWeight {
+			return errors.Wrap(errors.ErrInvalidInput, "must not be greater max weight")
+		}
+		if err := v.Address.Validate(); err != nil {
+			return errors.Wrapf(err, "address at position: %d", i)
+		}
+	}
+	return validateCreateProposal(&m)
+}
+
+type commonCreateProposalData interface {
+	GetTitle() string
+	GetDescription() string
+	GetElectorateID() []byte
+	GetStartTime() weave.UnixTime
+	GetAuthor() weave.Address
+}
+
+func validateCreateProposal(m commonCreateProposalData) error {
+	err := m.GetAuthor().Validate()
+	switch {
+	case len(m.GetElectorateID()) == 0:
+		return errors.Wrap(errors.ErrInvalidInput, "empty electorate id")
+	case m.GetStartTime() == 0:
+		return errors.Wrap(errors.ErrInvalidInput, "empty start time")
+	case m.GetAuthor() != nil && err != nil:
+		return errors.Wrap(err, "invalid author")
+	case !validTitle(m.GetTitle()):
+		return errors.Wrapf(errors.ErrInvalidInput, "title: %q", m.GetTitle())
+	case len(m.GetDescription()) < minDescriptionLength:
+		return errors.Wrapf(errors.ErrInvalidInput, "description length lower than minimum of: %d", minDescriptionLength)
+	case len(m.GetDescription()) > maxDescriptionLength:
+		return errors.Wrapf(errors.ErrInvalidInput, "description length exceeds: %d", maxDescriptionLength)
+	}
+	return m.GetStartTime().Validate()
```

Comment:
I think adding a wrapper explaining which field validation failed might be helpful

```go
if err := m.GetStartTime().Validate(); err != nil {
    return errors.Wrap(err, "start time")
}
```